soc/arm/microchip_mec/mec172x/device_power.h

```diff
@@ -22,7 +22,6 @@
 /* Enable SoC control over peripherals only when drivers do not support
  * power management
  */
-#ifndef CONFIG_PM_DEVICE
```

Comment:
It would be helpful to explain in the commit message why you're making this change. The comment above suggests that drivers now support power management?

Reply:
The change allows the SoC to enter deep sleep even when CONFIG_DEVICE_PM is selected to enable low power support for certain drivers like UART, ADC (which we are planning to add);
I will update the commit message accordingly; also will remove the comment (lines 22-24)

Reply:
Before this PR, the code went to the trouble of checking `ifndef CONFIG_PM_DEVICE`. Did something change that makes that check no longer necessary? Was it incorrect to do that check in the first place?

Reply:
Yes, it was incorrect in the first place. 
Our chip still requires the soc power file to perform some operations even when CONFIG_PM_DEVICE is enabled to allow the hardware to shut down the PLL.

Reply:
Please update the commit message with this explanation

Reply:
updated the commit